File: 3233-find-the-count-of-numbers-which-are-not-special/3233-find-the-count-of-numbers-which-are-not-special.py
class Solution:
    def nonSpecialCount(self, l: int, r: int) -> int:
        threshold = int(math.sqrt(r))
        
        prime = [True] * (threshold + 1)
        prime[0] = prime[1] = False
        
        for i in range(2, int(math.sqrt(threshold)) + 1):
            if prime[i]:
                for j in range(i * i, threshold + 1, i):
                    prime[j] = False
        
        count = 0
        for i in range(2, threshold + 1):
            if prime[i]:
                sq = i * i
                if l <= sq <= r:
                    count += 1
        
        total_numbers = r - l + 1
        
        return total_numbers - count

File: 3233-find-the-count-of-numbers-which-are-not-special/3233-find-the-count-of-numbers-which-are-not-special.cpp
class Solution {
public:
    int nonSpecialCount(int l, int r) {
        int threshold = sqrt(r);
        vector<int> prime(threshold+1, 1);
        prime[0] = 0, prime[1] = 0;
        for (int i = 2; i * i <= threshold; i++) {
            if (prime[i] == 1) {
                for (int j = i * i; j <= threshold; j += i) {
                    prime[j] = 0;
                }
            }
        }
        int count = 0;
        for (int i = 2; i <= threshold; i++) {
            if (prime[i] == 1) {
                int sq = i * i;
                if (sq >= l && sq <= r) {
                    count++;
                }
            }
        }
        return r - l + 1 - count;
    }
};